python/foglamp/common/configuration_manager.py

```diff
@@ -225,9 +233,12 @@ def get_entry_val(k):
 
                 # If Entry item exists in optional list, then update expected item entries
                 if entry_name in optional_item_entries:
-                    if entry_name == 'readonly':
+                    if entry_name == 'readonly' or entry_name == 'deprecated':
                         if self._validate_type_value('boolean', entry_val) is False:
                             raise ValueError('Unrecognized value for item_name {}'.format(entry_name))
+                        if entry_name == 'deprecated':
```

Comment:
why do we need this?

Reply:
bcoz we are restricting deprecated to true only.

Reply:
```
if entry_name == 'readonly' or entry_name == 'deprecated':
    if self._validate_type_value('boolean', entry_val) is False:
        raise ValueError('{} attribute can be set to [true/false] only'.format(entry_name))
```